changelogs/unreleased/gh-7746-fiber-channel-graceful-close.md

```diff
@@ -0,0 +1,6 @@
+## feature/core/fiber
```

Comment:
It would be nice to verify the new mode before we'll proceed with the patch. I collected a list of our modules that use channels.

----

Ones that I would verify in the new mode:

* tarantool/cartridge
* tarantool/moonwalker
* tarantool/kafka
* tarantool/mysql
* tarantool/pg
* tarantool/tracing

Where a channel is used in a test (but anyway interesting to check it):

* tarantool/expirationd
* tarantool/queue
* tarantool/sharded-queue
* tarantool/smtp
* tarantool/synchronized (beware, it is [flawed](https://github.com/tarantool/synchronized/issues/1))

Deprecated (but maybe worth to check):

* tarantool/pregel (no updates since 2016, but it is known to use `ch:close()`)

Deprecated (I would skip them):

* tarantool/zookeeper (last update in 2019, explicitly marked as community supported)
* tarantool/shard (deprecated several years ago)
* tarantool/sentry (archived)

----

My source data is two years old and I'm lazy to re-collect the repositories. I hope it gives a good test plan anyway.

You can ask me where to get help with each particular module.

Reply:
I set the default value of `compat.fiber_channel_close_mode` to `new` and run integration testing that covers cartridge, mysql, expirationd, queue. Anyway, I suggest to verify all external modules when we decide to change the default.